crates/autopilot/src/infra/persistence/mod.rs

```diff
@@ -374,6 +375,88 @@ impl Persistence {
 
         Ok(solution)
     }
+
+    pub async fn save_settlement(
+        &self,
+        event: domain::eth::Event,
+        auction: domain::auction::Id,
+        settlement: Option<&domain::settlement::Settlement>,
+    ) -> Result<(), error::Settlement> {
+        let _timer = Metrics::get()
+            .database_queries
+            .with_label_values(&["save_settlement"])
+            .start_timer();
+
+        let mut ex = self
+            .postgres
+            .pool
+            .begin()
+            .await
+            .map_err(error::Settlement::BadCommunication)?;
+
+        let block_number = event.block.0.try_into().unwrap(); // todo fix
+        let log_index = event.log_index.try_into().unwrap();
+
+        database::settlements::update_settlement_auction(&mut ex, block_number, log_index, auction)
+            .await
+            .map_err(error::Settlement::BadCommunication)?;
+
+        if let Some(settlement) = settlement {
+            let gas = settlement.gas();
+            let gas_price = settlement.gas_price();
+            let surplus = settlement.native_surplus();
+            let fee = settlement.native_fee();
+            let order_fees = settlement.order_fees();
+
+            tracing::debug!(
+                ?auction,
```

Comment:
nit: to align with other logs.
```suggestion
                auction_id = ?auction,
```